src/MuxManager.cpp

```diff
@@ -470,4 +472,79 @@ void MuxManager::generateServerMac(uint16_t serverId, std::array<uint8_t, ETHER_
     }
 }
 
+// ---> processWarmRestartFlag
+//
+// process warm restart flag
+//
+void MuxManager::processWarmRestartFlag(const std::string &flag)
+{
+    mWarmRestartFlag = flag;
+    
+    if(flag == "true") {
+        MUXLOGINFO("Detected a warm restart is on, starting reconciliation timer.");
+        startWarmRestartReconciliationTimer();
+    }
+}
+
+// ---> updateWarmRestartReconciliationCount(int increment);
+//
+// update warm restart reconciliation count
+//
+void MuxManager::updateWarmRestartReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(increment);
+
+    boost::asio::io_service &ioService = mStrand.context();
+
+    ioService.post(mStrand.wrap(boost::bind(
+        &MuxManager::handleUpdateReconciliationCount,
+        this,
+        increment
+    )));
+}
+
+// ---> handleUpdateReconciliationCount(int increment);
+//
+// handler of updating reconciliation port count
+// 
+void MuxManager::handleUpdateReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(mPortReconciliationCount);
+
+    mPortReconciliationCount += increment;
+
+    if(mPortReconciliationCount == 0) {
+        mReconciliationTimer.cancel();
+    } 
+}
+
+// ---> startWarmRestartReconciliationTimer
+//
+// start warm restart reconciliation timer
+// 
+void MuxManager::startWarmRestartReconciliationTimer()
+{
+    mReconciliationTimer.expires_from_now(boost::posix_time::seconds(
+        mMuxConfig.getMuxReconciliationTimeout_sec()
+    ));
+    mReconciliationTimer.async_wait(mStrand.wrap(boost::bind(
+        &MuxManager::handleWarmRestartReconciliationTimeout,
+        this,
+        boost::asio::placeholders::error
+    )));
+}
+
+// ---> handleWarmRestartReconciliationTimeout
+// 
+// handle warm restart reconciliationTimeout
+//
+void MuxManager::handleWarmRestartReconciliationTimeout(const boost::system::error_code errorCode)
+{
+    if (errorCode == boost::system::errc::success) {
```

Comment:
Just making sure - the errorCode in case of timeout will be == `success`?

Reply:
Correct.